### ISSUE TITLE
Make IReusableResult.Value public

### ISSUE THREAD

**Issue by drakepro** [DaveSkender/Stock.Indicators#953]
Loving the library so far.  I've been messing around with making my own indicators, but would like to more easily make them chainable.  I think it would be helpful if I could implement IReusableResult, but the Value property is currently internal.

public interface IReusableResult : IResult
{
    **internal** double? Value { get; }
} 

Could it be made public without changing what your intention was?

**Comment by DaveSkender** [comment]
There’s no technical harm to make it public; I’d only done that to avoid confusion, since it’s a fairly ambiguous without a descriptive name, and wasn’t thinking about its utility for custom indicators.

**Comment by DaveSkender** [comment]
Nice.  I can imagine you’re coming up with all sorts of interesting things.  We do have a general category for price characteristics / numerical analysis, so if you are using any of the classics, we could use more (e.g. Sharpe ratio, drawdown, etc.)

**Comment by DaveSkender** [comment]
For this issue, you’re looking to do something like this, right?

```csharp
public class AtrWmaResult : ResultBase, IReusableResult
{
    public double? AtrWma { get; set; }
    
    // to enable further chaining
    double? IReusableResult.Value => AtrWma;
}
```

**Comment by DaveSkender** [comment]
@mihakralj contributed a nice [random quote history generator](https://github.com/DaveSkender/Stock.Indicators/blob/4a26cb11baaf34060d01e5ee1b034fd6f5b04a7d/tests/indicators/_common/Helper.Random.cs) a while back.
Might be useful if you need something like that

**Comment by drakepro** [comment]
> For this issue, you’re looking to do something like this, right?
> 
> ```cs
> public class AtrWmaResult : ResultBase, IReusableResult
> {
>     public double? AtrWma { get; set; }
>     
>     // to enable further chaining
>     double? IReusableResult.Value => AtrWma;
> }
> ```

That's right.  

I'll keep an eye out for anything I can contribute that might be usable for others.

**Comment by DaveSkender** [comment]
This change is relatively straightforward, so I'll try to get to it in the next week or so (I will try this weekend).  This change has some potentially good impact on the extensibility of results classes overall; however, I have to test a few things to ensure good UX.

If you're feeling ambitious and want this sooner, [fork the repo](https://github.com/DaveSkender/Stock.Indicators/discussions/503) and [contribute](https://dotnet.stockindicators.dev/contributing/#content) to the feature branch depicted in the PR #954.

**Comment by DaveSkender** [comment]
This was released in v2.4.2
